src/pipelines/_common/wikidata.py

```diff
@@ -31,12 +31,12 @@ def _process_item(props: Dict[str, str], key_wikidata: Tuple[str, str]):
         return {"key": key, **wikidata_properties(props, wikidata_id)}
 
     def parse(self, sources: Dict[str, str], aux: Dict[str, DataFrame], **parse_opts) -> DataFrame:
-        data = aux["knowledge_graph"].merge(aux["metadata"])[["key", "wikidata"]].set_index("key")
+        data = aux["knowledge_graph"].merge(aux["metadata"])[["key", "wikidata"]]
 
         # Load wikidata using parallel processing
-        map_iter = data.wikidata.iteritems()
         map_func = partial(self._process_item, parse_opts)
-        records = thread_map(map_func, map_iter, total=len(data), desc="Wikidata Properties")
+        map_iter = data.dropna().set_index("key")["wikidata"].iteritems()
+        records = thread_map(map_func, list(map_iter), desc="Wikidata Properties")
```

Comment:
Is the list(map_iter) here necessary?

Reply:
It is used so `thread_map` knows how many items there are and it can provide a progress bar. The alternative is to count how many items are left after `data.dropna()` and pass that to the `total` parameter. Since there are <20K items in `map_iter`, the effect on performance appears to be negligible.

That said, I *really* want to improve this whole file. I think using the JSON API is suboptimal and we are apparently not always grabbing the latest known values (sometimes we are using population counts which are several years old). I think using SPARQL can help improve the performance and reliability of the data, and I recall you had some CL a while back which was using SPARQL to pull data but I don't think you ever submitted it. Do you think you could help refactor this file to use your code to submit a SPARQL query?